### PR TITLE
raftstore: Increase the default raft_client_queue_size and raft_msg_max_batch_size.

### LINKED ISSUE #17101: Commit log duration is high when hibernate regions are awake
## Bug Report

b57e15ae-443b-47de-867e-033bfeb70713

### PR DESCRIPTION
**Pull request by hhwyt**
…raft_msg_max_batch_size

### What is changed and how it works?

Issue Number: Close #17101

What's Changed:

```commit-message
Increase the default raft_client_queue_size and raft_msg_max_batch_size.

This PR addresses an issue where too many Raft messages can delay 
sending, increasing the commit log duration and the heartbeat latency. 
The delayed heartbeats can lead to leader drops, especially during PD 
restarts that trigger a surge of hibernated regions. About this scenario, 
see more details at: https://github.com/tikv/tikv/issues/17101.

We increased the raft_client_queue_size to prevent Raft messages from 
being dropped when the RaftClient queue becomes full under too many 
message workloads. Additionally, we increased the raft_msg_max_batch_size 
to improve the efficiency of Raft message sending.
```

### Related changes

- [x] PR to update `pingcap/docs`/`pingcap/docs-cn`:
  - https://github.com/pingcap/docs/pull/19454
  - https://github.com/pingcap/docs-cn/pull/19120
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
increase the default raft-client-queue-size from 8192 to 16384.
```

### CODE REVIEW

**Comment by hhwyt** [comment]
No benchbot due to limited resources, hope to meet the 8.5 code freeze.

**Review by LykxSassinator — APPROVED** [review]

File: src/server/config.rs [hunk 1]
```diff
@@ -256,10 +256,10 @@ impl Default for Config {
             status_thread_pool_size: 1,
             max_grpc_send_msg_len: DEFAULT_MAX_GRPC_SEND_MSG_LEN,
             raft_client_grpc_send_msg_buffer: 512 * 1024,
-            raft_client_queue_size: 8192,
+            raft_client_queue_size: 16384,
```
Comment (LykxSassinator) [inline src/server/config.rs:259]: nit: please add some annotations here to specify that the default value has been enlarged by this pr.

Comment (hhwyt) [reply src/server/config.rs:259]: In what situations would we care when this default value is increased, requiring us to look this PR?

Comment (hhwyt) [reply src/server/config.rs:259]: If there are no such situations, we don't need an annotation here.

Comment (LykxSassinator) [reply src/server/config.rs:259]: Better to leave this note as a reminder.

Comment (hhwyt) [reply src/server/config.rs:259]: okay

Comment (hhwyt) [reply src/server/config.rs:262]: done

**Review by glorv — APPROVED** [review]

**Comment by hhwyt** [comment]
@zhangjinpeng87 This PR requires higher-level permissions. PTAL, thx~

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**